pkg/kubectl/util/hash/hash.go

```diff
@@ -56,7 +56,11 @@ func SecretHash(sec *v1.Secret) (string, error) {
 // Data, Kind, and Name are taken into account.
 func encodeConfigMap(cm *v1.ConfigMap) (string, error) {
 	// json.Marshal sorts the keys in a stable order in the encoding
-	data, err := json.Marshal(map[string]interface{}{"kind": "ConfigMap", "name": cm.Name, "data": cm.Data})
+	m := map[string]interface{}{"kind": "ConfigMap", "name": cm.Name, "data": cm.Data}
+	if len(cm.BinaryData) > 0 {
+		m["binaryData"] = cm.BinaryData
```

Comment:
does the stdlib marshal []byte in unstructured maps correctly?

Reply:
yes, see `TestEncodeConfigMap` method `"three keys"` variation. uses base64 encode

Reply:
https://golang.org/pkg/encoding/json/#Marshal
>Array and slice values encode as JSON arrays, except that []byte encodes as a base64-encoded string